### PR TITLE
LibJIT+LibJS+Embedders: Lazily collect stack traces

### PR DESCRIPTION
**Pull request by skyrising**
The previous implementation was calling `backtrace()` for every function call, which is quite slow.
I noticed while running `Kraken/imaging-gaussian-blur` which took ages.
With these changes it's back to ~20s on my machine.

### CODE REVIEW

File: Userland/Applications/Spreadsheet/Spreadsheet.cpp [hunk 1]
```diff
@@ -72,11 +72,7 @@ Sheet::Sheet(Workbook& workbook)
                 if (thrown_value.is_object() && is<JS::Error>(thrown_value.as_object())) {
                     auto& error = static_cast<JS::Error const&>(thrown_value.as_object());
                     warnln(" with message '{}'", error.get_without_side_effects(vm.names.message));
-                    for (auto& traceback_frame : error.traceback()) {
-                        auto& function_name = traceback_frame.function_name;
-                        auto& source_range = traceback_frame.source_range();
-                        dbgln("  {} at {}:{}:{}", function_name, source_range.filename(), source_range.start.line, source_range.start.column);
-                    }
+                    dbgln(error.stack_string(JS::CompactTraceback::Yes));
```
Comment (alimpfard) [inline Userland/Applications/Spreadsheet/Spreadsheet.cpp:75]: Please do not pass a random string as the format parameter.
```suggestion
                    dbgln("{}", error.stack_string(JS::CompactTraceback::Yes));
```

File: Userland/Libraries/LibWeb/HTML/Scripting/ExceptionReporter.cpp [hunk 2]
```diff
@@ -31,11 +31,7 @@ void report_exception_to_console(JS::Value value, JS::Realm& realm, ErrorInPromi
         }
         if (is<JS::Error>(object)) {
             auto const& error_value = static_cast<JS::Error const&>(object);
-            for (auto& traceback_frame : error_value.traceback()) {
-                auto& function_name = traceback_frame.function_name;
-                auto& source_range = traceback_frame.source_range();
-                dbgln("  {} at {}:{}:{}", function_name, source_range.filename(), source_range.start.line, source_range.start.column);
-            }
+            dbgln(error_value.stack_string(JS::CompactTraceback::Yes));
```
Comment (alimpfard) [inline Userland/Libraries/LibWeb/HTML/Scripting/ExceptionReporter.cpp:34]: Same here.

File: Userland/Utilities/js.cpp [hunk 3]
```diff
@@ -251,36 +251,7 @@ static ErrorOr<bool> parse_and_run(JS::Realm& realm, StringView source, StringVi
...
-                            traceback_frame.function_name,
-                            traceback_frame.source_range().code->filename(),
-                            traceback_frame.source_range().start.line,
-                            traceback_frame.source_range().start.column);
-                    }
-                }
-                repetitions = 0;
-            }
-        }
+        warn(static_cast<JS::Error const&>(thrown_value.as_object()).stack_string(JS::CompactTraceback::Yes));
```
Comment (alimpfard) [inline Userland/Utilities/js.cpp:254]: And here.

**Comment by awesomekling** [comment]
This has conflicts

**Comment by ADKaster** [comment]
(conflicts that are my fault, sorry!)

**Pull request merged**